Library/Util.ts

```diff
@@ -146,33 +146,77 @@ class Util {
     }
 
     /**
-     * Validate that an object is of type { [key: string]: string }
+     * Using JSON.stringify, by default Errors do not serialize to something useful:
+     * Simplify a generic Node Error into a simpler map for customDimensions
+     * Custom errors can still implement toJSON to override this functionality
      */
-    public static validateStringMap(obj: any): { [key: string]: string } {
-        var map: { [key: string]: string };
-        if(typeof obj === "object") {
-            map = <{ [key: string]: string }>{};
-            for (var field in obj) {
-                var property = obj[field];
-                var propertyType = typeof property;
-                if (propertyType !== "string") {
-                    if (property != null && typeof property.toString === "function") {
-                        property = property.toString();
-                    } else {
-                        Logging.info("key: " + field + ", invalid property type: " + propertyType);
-                        continue;
-                    }
-                }
+    protected static extractError(err: Error): { message: string, stack_trimmed: string } {
```

Comment:
It might be better to reuse the stack parsing that already exists in this SDK (see https://github.com/Microsoft/ApplicationInsights-node.js/blob/develop/Library/EnvelopeFactory.ts#L242 )

If we don't reuse, we should ensure the `stack` object exists and is a string. Node code can use the `prepareStackTrace` method to change the construction of the `stack` object (see https://github.com/v8/v8/wiki/Stack-Trace-API#customizing-stack-traces).

That being said, I'm not sure more than message should be recorded to avoid reaching the quite short length limits on custom dimensions.

Reply:
OK - I'll drop the stack.